RELEASE.md

```diff
@@ -1,39 +1,51 @@
 # Releasing the Jaeger Operator for Kubernetes
 
-1. Update Jaeger version in `versions.txt`
+Steps to release a new version of the Jaeger Operator:
 
-1. Make sure the new version is present at `pkg/upgrade/versions.go`
 
-1. Prepare a changelog since last release. Get the `OAUTH_TOKEN` from (Github)[https://github.com/settings/tokens/new?description=GitHub%20Changelog%20Generator%20token] and select `repo:status` scope.
+1. Change the `versions.txt `so that it lists the target version of the Jaeger (if it is required). Don't touch the operator version it will be changed automatrically in the next ste.
 
-    ```
-    OAUTH_TOKEN=... make changelog
-    ```
+2. Run `make prepare-release VERSION=1.25.0`, using the operator version that will be released.
+
+3. Prepare a changelog since last release. 
 
-1. Commit version change and changelog and create a pull request:
+4. Commit the changes and create a pull request:
 
    ```
    git commit -sm "Preparing release v1.25.0"
    ```
 
-1. Tag and push
+5. Once the changes above are merged and available in `master` tag it with the desired version, prefixed with `v`, eg. `v1.25.0`
 
     ```
-    git checkout master ## it's only possible to release from master for now!
-    git tag release/v1.25.0
-    git push git@github.com:jaegertracing/jaeger-operator.git release/v1.25.0
```

Comment:
any reason why the tag name is changed?

Reply:
The reason is this:
 - In the previous process for trigger a release we push a tag with `release/v1.25.0`, this triggers the process , the old process creates the manifest and push it to master and creates the `v1.25.0 `tag
 - In the new process I proposed here, you send a PR with the `CHANGELOG.md`, the new manifest and the `versions.txt` changes. when the PR is merged you only need to push a tag it with `v1.25.0 `. I'm skipping one step, the step of tagging with `release/v1.25.0`

Does that makes sense?

Reply:
This line changes the `versions.txt `: https://github.com/jaegertracing/jaeger-operator/pull/1535/files#diff-84879d57aeb7ca56c4e920fb01240d818d775283ebbcc2860c347d4507115a4cL39

I can do that, but involves more steps, more steps more prone to error.  (specially more automatic things that could fail) IMHO.

Reply:
WDYT? @jpkrohling  @pavolloffay

Reply:
Agree with @pavolloffay that the release process should instead just open a PR with the changes. A tag can then be created based on that. So, we are back at the `release/v.X.Y.Z` to trigger the release and `vX.Y.Z` with the release itself.

Reply:
We talked after I left my previous comment, and I agree with the simplified approach: run `make prepare-release VERSION=1.25.0`, open the PR, merge, then tag with `v1.25.0` directly, without the intermediary tag.